cloudfront.tf

```diff
@@ -0,0 +1,90 @@
+resource "aws_cloudfront_distribution" "this" {
+  comment = aws_s3_bucket.this.bucket
+  enabled = true
+
+  aliases = var.aliases
+
+  http_version = "http2"
+
+  origin {
+    origin_id   = "S3-${aws_s3_bucket.this.bucket}"
+    domain_name = aws_s3_bucket.this.bucket_domain_name
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.this.cloudfront_access_identity_path
+    }
+  }
+
+  default_cache_behavior {
+    target_origin_id       = "S3-${aws_s3_bucket.this.bucket}"
+    viewer_protocol_policy = "https-only"
+
+    allowed_methods = [
+      "GET",
+      "HEAD",
+    ]
+
+    cached_methods = [
+      "GET",
+      "HEAD",
+    ]
+
+    forwarded_values {
+      query_string = false
+
+      headers = [
+        "Access-Control-Request-Headers",
+        "Access-Control-Request-Method",
+        "Origin",
+      ]
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    min_ttl     = var.ttl.min
+    default_ttl = var.ttl.default
+    max_ttl     = var.ttl.max
+
+    compress = true
+  }
+
+  viewer_certificate {
+    acm_certificate_arn            = local.viewer_certificate.acm_certificate_arn
+    cloudfront_default_certificate = local.viewer_certificate.cloudfront_default_certificate
+
+    minimum_protocol_version = local.viewer_certificate.minimum_protocol_version
+    ssl_support_method       = local.viewer_certificate.ssl_support_method
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  tags = var.tags
+}
+
+resource "aws_cloudfront_origin_access_identity" "this" {
+  comment = aws_s3_bucket.this.bucket
+}
+
+locals {
+  default_viewer_certificate = {
+    acm_certificate_arn            = null
+    cloudfront_default_certificate = true
+    minimum_protocol_version       = "TLSv1"
+    ssl_support_method             = null
+  }
+
+  acm_viewer_certificate = {
+    acm_certificate_arn            = var.acm_certificate_arn
+    cloudfront_default_certificate = false
+    minimum_protocol_version       = "TLSv1"
```

Comment:
Let's check whether other values for custom certificates are supposed by now.

Reply:
@jansiwy According to the table [here](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/secure-connections-supported-viewer-protocols-ciphers.html), there are more options now. The latest requires TLSv1.3.

Which is one advisable here?

Reply:
I'd recommend to make it configurable, but use `TLSv1.3` as default. When we migrate to this module, we'd set the TLS version we currently use and create another card to update the TLS version everywhere.

Let's please also check first whether both cases, `default_viewer_certificate` and `acm_viewer_certificate` actually allow to configure the TLS version. In the past, it was not configurable for `default_viewer_certificate`.

Reply:
@jansiwy From the docs for this resource:

> If you have specified `cloudfront_default_certificate`, `TLSv1` must be specified.

So we have to continue using branching here, depending on whether we use the default viewer cert.

Reply:
fixed in [2405a5b](https://github.com/babbel/terraform-aws-cloudfront-bucket/pull/1/commits/2405a5b74a7a5db60589b2d9daa0a798a78a6c0c)